From fcda9c6a208d3a7fe651ef661b2eb6e462a89c17 Mon Sep 17 00:00:00 2001
From: Eneas U de Queiroz <cote2004-github@yahoo.com>
Date: Tue, 31 Jul 2018 17:44:57 -0300
Subject: [PATCH] CMake: use find_package (SWIG) for cmake >= 3.0.0

There's a workaround for a bug (fixed in cmake 3.0.0) that does not
work in CMake 3.12.  Only use the workaround with cmake < 3.0.0.

Signed-off-by: Eneas U de Queiroz <cote2004-github@yahoo.com>
[Mark: patch retrieved and updated from
http://developer.intra2net.com/git/?p=libftdi;a=commit;h=fcda9c6a208d3a7fe651ef661b2eb6e462a89c17]
Signed-off-by: Mark Corbin <mark.corbin@embecosm.com>
[Update patch to make it work with cmake < 3.7:
http://developer.intra2net.com/mailarchive/html/libftdi/2019/msg00009.html]
Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
---
diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index 31ef1c6..eefc344 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -3,7 +3,7 @@ option ( LINK_PYTHON_LIBRARY "Link against python libraries" ON )
 
 if ( PYTHON_BINDINGS )
   # workaround for cmake bug #0013449
-  if ( NOT DEFINED CMAKE_FIND_ROOT_PATH )
+  if ( NOT DEFINED CMAKE_FIND_ROOT_PATH OR NOT CMAKE_VERSION VERSION_LESS 3.0.0 )
     find_package ( SWIG )
   else ()
     find_program ( SWIG_EXECUTABLE NAMES swig2.0 swig )
